test/testdata/resolver/flatten.rb.symbol-table-raw.exp

```diff
@@ -17,13 +17,6 @@ class <C <U <root>>> < <C <U Object>> () @ (... removed core rbi locs ..., Loc {
       argument <blk><block> @ Loc {file=test/testdata/resolver/flatten.rb start=??? end=???}
     method <S <C <U A>> $1><U sfood> (<blk>) @ Loc {file=test/testdata/resolver/flatten.rb start=10:5 end=10:14}
       argument <blk><block> @ Loc {file=test/testdata/resolver/flatten.rb start=??? end=???}
-  class <S <S <C <U A>> $1> $1>[<C <U <AttachedClass>>>] < <S <S <C <U Object>> $1> $1> () @ Loc {file=test/testdata/resolver/flatten.rb start=2:7 end=2:8}
-    type-member(+) <S <S <C <U A>> $1> $1><C <U <AttachedClass>>> -> LambdaParam(<S <S <C <U A>> $1> $1><C <U <AttachedClass>>>, lower=T.noreturn, upper=AppliedType {
-  klass = <S <C <U A>> $1>
-  targs = [
-    <C <U <AttachedClass>>> = A
-  ]
-}) @ Loc {file=test/testdata/resolver/flatten.rb start=2:7 end=2:8}
-    method <S <S <C <U A>> $1> $1><U sfoos> (<blk>) @ Loc {file=test/testdata/resolver/flatten.rb start=12:5 end=12:19}
+    method <S <C <U A>> $1><U sfoos> (<blk>) @ Loc {file=test/testdata/resolver/flatten.rb start=12:5 end=12:19}
```

Comment:
this is now a member of a different class

Reply:
The changes here help us fix #122. The relevant parts of the test file in question look like this:
```ruby
class A
  def self.sfoo
    def self.sfoos; end
  end
end
```
Previously, we treated the definition of `sfoos` as though it were somehow being added to `T.class_of(T.class_of(A))`. After these changes, we now treat it correctly as a static method on `A` just like `sfoo`.

Reply:
Is it possible to separate the behavior change into a separate PR?

Reply:
I'll look into it: it's possible, but I don't know off-hand how easy or difficult it would be. The issue is that I can't make the syntactic flattening pass produce methods that are defined on `T.class_of(T.class_of(A))`, because there's no sensible way to create a Ruby AST that does that. So the only way to land that bugfix first is by fixing it in the namer, and then landing this PR that makes the other bugfix redundant: I'll look into how much work that would be.

Reply:
> because there's no sensible way to create a Ruby AST that does that.

```
class A
  class <<self
    class <<self
      def foo; end
    end
  end
end
```

Reply:
or, equivalently
```
class A
  class <<self
      def self.foo; end
  end
end
```